src/documents/CoreAPI.md

```diff
@@ -0,0 +1,675 @@
+
+* `tocopy {Object|Array}`
+* Returns: `{Object|Array}`
+
+Return an empty container as the same type as the argument (either an array or hash). To be wise, the supplied argument should 
+pass the `fluid.isPlainObject` test.
+
+### fluid.copy(tocopy)
+
+* `tocopy {Any}`
+* Returns: `{Any}`
+
+Performs a deep copy (clone) of its argument. This will guard against cloning a circular object by terminating if it reaches a path depth
+greater than [`fluid.strategyRecursionBailout`](#fluid-strategyrecursionbailout). `fluid.copy` will not copy an object which passes `fluid.isUncopyable`.
+
+### fluid.makeArray(arg)
+
+* `arg {Any}`
+* Returns: `{Array}`
+
+Converts the supplied argument to an array (or copies it if it is already an array), by a variety of strategies:
+
+* If it is even slightly arrayable (has a numeric `length` property), its elements will be copied into a fresh array
+* If it is not slightly arrayable, it will be placed into a 1-element array
+* If it is undefined, a 0-element array will be returned
+
+### fluid.generate(n, generator[, applyFunc])
+
+* `n {Integer}` The size of the array to be filled
+* `generator {Object|Function}` Either a value to be replicated or function to be called (if the `applyFunc` argument holds `true`). In case of a function, the signature is `(i {Integer}) → Any`, accepting the index position and returning the generated element.
+* `applyFunc {Boolean}` If `true`, the `generator` value is to be treated as a function
+* Returns: `{Array}` An array of length `n` holding the generated elements
+
+Fills an array of given size with copies of a value or result of a function invocation
+
+### fluid.iota(n[, first])
+
+* `n {Integer}` Size of the filled array to be returned
+* `first {Number}` [optional, defaults to 0] First element to appear in the array
+* Returns: `{Array}` An array of length `n` holding the generated elements
+
+Returns an array of size `n`, filled with increasing integers, starting at 0 or at the value specified by `first`. The name `iota` originally stems from the 
+[APL](http://stackoverflow.com/questions/9244879/what-does-iota-of-stdiota-stand-for) programming language.
+
+
+### fluid.clear(target)
+
+* `target {Object|Array}` The target to be cleared
+* Returns: `{Undefined}`
+
+Clears an object or array of its contents. For objects, each property is deleted. *** The input argument will be destructively modified ***
+
+
+## Transforming and filtering
+
+### fluid.transform(source, [fn1, fn2, ...])
+
+Return a list or hash of objects, transformed by one or more functions. Similar to
+[`jQuery.map`](http://api.jquery.com/jquery.map/), but accepts an arbitrary list of transformation functions and does not explode on null sources. 
+`fluid.transform` on a non-array and non-object (e.g. `null`) is a no-op.
+
+* `source {Array|Object}` The initial container of objects to be transformed. If the source is neither an array nor an object, it will be returned untransformed
```

Comment:
Please pick one format for all these API descriptions – either text description before or after the arguments description – and stick to it, on this page an any others.